pkg/tfcli/error.go

```diff
@@ -0,0 +1,30 @@
+package tfcli
+
+import "fmt"
+
+// OperationType is an operation type for terraform cli
+type OperationType int
+
+const (
+	OperationUnknown OperationType = iota
+	OperationObserve
+	OperationCreate
+	OperationUpdate
+	OperationDelete
+)
+
+func (o OperationType) String() string {
+	return []string{"Unknown", "Observe", "Create", "Update", "Delete"}[o]
+}
+
+type OperationInProgressError struct {
```

Comment:
Actually, we discussed this option with @ulucinar and finally agreed that it makes more sense to it this way since trying to kick an operation while another one is in progress is an error case from tfcli point of view. Handling this on the upper layer (cli adapter) sounds more reasonable. In other words, it is a valid state for controller/reconciler but not a valid one for tfcli.

Reply:
Yes, I agree. We would not like to allow concurrent Terraform operations (technically commands)  in the same Terraform workspace. From `tfcli`'s PoV it's an error to request a new operation before an ongoing one completes. This is also in sync with Terraform CLI's behaviour: if an ongoing Terraform command holds the state lock and we try to run a new command, the latter command cannot acquire the lock and fails. Thus, the proposal was to return an error to the caller, who wants to initiate a new operation via `tfcli`, if there is an already ongoing operation. And similar to `k8s.io/apimachinery/pkg/api/errors` package, provide utilities to deduce error types so that the caller can take appropriate actions.

Reply:
Sounds good!